client/llb/state.go

```diff
@@ -283,6 +283,13 @@ func (s State) File(a *FileAction, opts ...ConstraintsOpt) State {
 		o.SetConstraintsOption(&c)
 	}
 
+	// No platform was set by constraint options. Set the current OS and arch.
+	if c.Platform == nil {
+		c.Platform = &ocispecs.Platform{
+			OS:           "linux",
+			Architecture: "amd64",
```

Comment:
Is the architecture in here used anywhere? If not then I would prefer if the nil case is handled inside `FileOp` for `OS` only and it is clear that this only controls the path conversions.

Reply:
Removed the `Architecture`.

Reply:
What I meant was to handle nil platform inside FileOp. If the logic is only based on OS then we shouldn't control it by passing full platform (or passing an invalid partial platform).

Reply:
Gotcha. Misread what you meant.